modules/tinymce/src/core/test/ts/browser/selection/MultiClickSelectionTest.ts

```diff
@@ -1,32 +1,128 @@
+      assertRange(expectedRng, findClosestBlockRange(startRng, getRoot()));
+
+    it(`should return the range with the whole text content of the block`, () => {
+      setupHtml('<p>aaa bIb ccc</p>');
+      const textNode = getRoot().firstChild.firstChild;
+      const startRng = createRange(textNode, 4, textNode, 7);
+      const expectedRng = createRange(textNode, 0, textNode, 11);
+      testFindClosestBlockRange(startRng, expectedRng);
+    });
+
+    it(`should return the range expanded from the block start to the next closest <br> element`, () => {
+      setupHtml('<p>aaa bIb <strong>ccc</strong><br>ddd</p>');
+      const firstTextNode = getRoot().firstChild.firstChild;
+      const cccTextNode = getRoot().firstChild.childNodes[1].firstChild;
+      const startRng = createRange(firstTextNode, 3, firstTextNode, 6);
+      const expectedRng = createRange(firstTextNode, 0, cccTextNode, 3);
+      testFindClosestBlockRange(startRng, expectedRng);
+    });
+
+    it(`should return the range expanded from <br> element to end of the block`, () => {
+      setupHtml('<p>aaa<br><em>bbb</em> cIc ddd</p>');
+      const lastTextNode = getRoot().firstChild.lastChild;
+      const bbbTextNode = getRoot().firstChild.childNodes[2].firstChild;
+      const startRng = createRange(lastTextNode, 1, lastTextNode, 4);
+      const expectedRng = createRange(bbbTextNode, 0, lastTextNode, 8);
+      testFindClosestBlockRange(startRng, expectedRng);
+    });
+
+    it(`should return the range expanded between the two <br> elements`, () => {
+      setupHtml('<p>aaa<br><em>bbb</em> cIc <strong>ddd</strong><br>eee</p>');
+      const clickTextNode = getRoot().firstChild.childNodes[3];
+      const bbbTextNode = getRoot().firstChild.childNodes[2].firstChild;
+      const dddTextNode = getRoot().firstChild.childNodes[4].firstChild;
+      const startRng = createRange(clickTextNode, 0, clickTextNode, 5);
+      const expectedRng = createRange(bbbTextNode, 0, dddTextNode, 3);
+      testFindClosestBlockRange(startRng, expectedRng);
+    });
+
+    it(`should return range expanded beween the block start and <br> elememt when the start range is exacly the <br> element`, () => {
+      setupHtml('<p>aaa bbb ccc<br>ddd</p>');
+      const pNode = getRoot().firstChild;
+      const firstTextNode = getRoot().firstChild.firstChild;
+      const startRng = createRange(pNode, 1, pNode, 2); // select exactly <br> element
+      const expectedRng = createRange(firstTextNode, 0, pNode, 2);
+      testFindClosestBlockRange(startRng, expectedRng);
+    });
+  });
+
+  describe('fake multi click browser test', () => {
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      base_url: '/project/tinymce/js/tinymce'
+    }, [], true);
+
+    const fakeMultiClick = (editor: Editor, clickCount: number, target: HTMLElement) => {
+      editor.dispatch('mousedown', { detail: clickCount, target } as unknown as MouseEvent);
```

Comment:
I wonder if we should add details as to mouse down settings in Agar Mouse so that it's more of a simulated real even than a simulated tinymce event.